maven/lib-gwt.sh

```diff
@@ -64,9 +64,16 @@ function maven-gwt() {
   GWT_EXTRACT_DIR=`ls $RANDOM_DIR | tail -n1`
   GWT_EXTRACT_DIR=$RANDOM_DIR/$GWT_EXTRACT_DIR
 
-  JAVADOC_FILE_PATH=$RANDOM_DIR/gwt-javadoc.jar #TODO also wrong for jakarta
+  JAVADOC_FILE_PATH=$RANDOM_DIR/gwt-javadoc.jar
   [ -d $GWT_EXTRACT_DIR/doc/javadoc ] && jar cf $JAVADOC_FILE_PATH -C $GWT_EXTRACT_DIR/doc/javadoc .
 
+  JAVADOC_JAKARTA_FILE_PATH=$RANDOM_DIR/gwt-jakarta-javadoc.jar
+  if [[ -d $GWT_EXTRACT_DIR/doc/javadoc ]]; then
+      cp -r $GWT_EXTRACT_DIR/doc/javadoc $GWT_EXTRACT_DIR/doc/javadoc-jakarta
+      find $GWT_EXTRACT_DIR/doc/javadoc-jakarta -type f -exec sed -i 's/javax.servlet/jakarta.servlet/g' {} +
+      jar cf $JAVADOC_JAKARTA_FILE_PATH -C $GWT_EXTRACT_DIR/doc/javadoc-jakarta .
```

Comment:
Good point with the javadoc homepage. I understood that is only needed as jar. 

In my understanding, that changing the package that would break the downstream compatibility.  But of course, it would simplify the support. For me, it is fine to change the package. The package name changing to jakarta force me anyway to make some code change. But it should be explicit in the gwt code and not a string replacement during the build.

Reply:
I _think_ we can get away with some string replacement in ant rather than copying and slightly changing files in git. I did something like this when I was helping grpc-java support both javax and jakarta servlets in https://github.com/grpc/grpc-java/pull/8596/files#diff-043e57914011bd202661fd70d29a97e7f3fdba84b2e6de7eace184707878fc8bR40-R53. We will have to restrict ourselves to very specific changes though.

As far as "breaking" downstream projects I think that is actually a safe thing to do here, since this is a breaking change anyway - any RemoteServiceServlet subclass that is already using any servlet features of any kind will already need to break in this way, right? We don't need to change the RemoteService interface at all, just the specific classes that interact with servlets, and any other classes that depend on those classes, etc.

This would protect projects that accidentally leave any old jars on the classpath (gwt-dev.jar is a very common jar in debugging to leave around) from running the risk of picking up the javax version instead of the jakarta version.

What do you think?

Reply:
Sorry for my delayed answer. 

For me, it is fine when string replacement is the way to go. Copying files make it explicit, but I understand your point. Your are the maintainer, so you decide :)

I will adjust the PR next week to fix the javadoc issue.